### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by TheSlimvReal**
Includes #9

This PR implements the user authentication.

- [x] Provides the session endpoint to which the user first has to authenticate. On success a JWT token is returned as a cookie. Authentication is currently done against CouchDB but this could be swapped out to another provider.
- [x] All CouchDB replication endpoints require a valid JWT token
- [x] The JWT token holds (encrypted) information about name and roles of the user. This makes the server stateless and allows multiple users to send simultaneous requests.

Missing:

- [x] Currently the JWT token has a fixed lifetime (60s). Under which circumstances should the token be refreshed? @sleidig

### CODE REVIEW

File: src/app.module.ts [hunk 1]
```diff
@@ -2,10 +2,13 @@ import { Module } from '@nestjs/common';
...
 
 @Module({
-  imports: [HttpModule],
+  imports: [HttpModule, SessionModule],
   controllers: [CouchProxyController],
   providers: [DocumentFilterService],
 })
 export class AppModule {}
+
+export const COUCH_ENDPOINT = 'https://dev.aam-digital.com/db';
```
Comment (sleidig) [inline src/app.module.ts:14]: Let's use environment variables on the server-side services (probably using dotenv https://nodejs.dev/learn/how-to-read-environment-variables-from-nodejs)

File: src/couch-proxy/couch-proxy.controller.ts [hunk 2]
```diff
@@ -23,44 +24,29 @@ import {
...
-@Controller('couchdb/db')
+@UseGuards(JwtGuard)
+@Controller()
 export class CouchProxyController {
-  readonly couchDB = 'https://dev.aam-digital.com/db';
-  private username: string;
-  private password: string;
-  public userRoles: string[] = ['user'];
+  private username: string = 'demo';
+  private password: string = 'pass';
```
Comment (sleidig) [inline src/couch-proxy/couch-proxy.controller.ts:35]: do we still need these at all?

**Comment by sleidig** [comment]
In my opinion the server login should not expire any time while a user is having the app open. So probably at least within 1-2 hours of inactivity users should get an automatically refreshed token? Once the server login expires we should also log out in the local app (and that should rarely happen during a workday, otherwise we'll drive users crazy)

File: src/app.module.ts (hunk 1 above)
Comment (TheSlimvReal) [reply src/app.module.ts]: #4

**Pull request merged**